MetaMorpheus/EngineLayer/FdrAnalysis/PEPValueAnalysisGeneric.cs

```diff
@@ -87,113 +153,173 @@ public static string ComputePEPValuesForAllPSMsGeneric(List<SpectralMatch> psms,
                 {
-                    fileSpecificTimeDependantHydrophobicityAverageAndDeviation_unmodified = ComputeHydrophobicityValues(peptides, fileSpecificParameters, false);
-                    fileSpecificTimeDependantHydrophobicityAverageAndDeviation_modified = ComputeHydrophobicityValues(peptides, fileSpecificParameters, true);
-                    fileSpecificTimeDependantHydrophobicityAverageAndDeviation_CZE = ComputeMobilityValues(peptides, fileSpecificParameters);
+                    decoyIndices.Add(i);
                 }
                 else
                 {
-                    fileSpecificTimeDependantHydrophobicityAverageAndDeviation_unmodified = ComputeHydrophobicityValues(psms, fileSpecificParameters, false);
-                    fileSpecificTimeDependantHydrophobicityAverageAndDeviation_modified = ComputeHydrophobicityValues(psms, fileSpecificParameters, true);
-                    fileSpecificTimeDependantHydrophobicityAverageAndDeviation_CZE = ComputeMobilityValues(psms, fileSpecificParameters);
+                    targetIndices.Add(i);
                 }
             }
 
-            if (trainingVariables.Contains("ChimeraCount"))
-                chimeraCountDictionary = psms.GroupBy(p => p.ChimeraIdString)
-                    .ToDictionary(p => p.Key, p => p.Count());
-
-            MLContext mlContext = new MLContext();
+            var targetIndexGroups = DivideListIntoGroups(targetIndices, numGroups);
+            var decoyIndexGroups = DivideListIntoGroups(decoyIndices, numGroups);
 
-            //the number of groups used for cross-validation is hard-coded at four. Do not change this number without changes other areas of effected code.
-            int numGroups = 4;
-            if (psms.Count < 1000 || allPeptideIndices.Count < 500)
+            for (int i = 0; i < numGroups; i++)
             {
-                numGroups = 2;
+                groupsOfIndices[i] = targetIndexGroups[i].Concat(decoyIndexGroups[i]).ToList();
             }
-            List<int>[] psmGroupIndices = Get_PSM_Group_Indices(psms, numGroups);
 
-            //the psms will be randomly divided. but then we want to make another array that just contains the subset of peptides that are in those psms. that way we don't compute pep using any peptides that were used in training.
-            List<int>[] peptideGroupIndices = Get_Peptide_Group_Indices(psmGroupIndices, allPeptideIndices);
-            IEnumerable<PsmData>[] PSMDataGroups = new IEnumerable<PsmData>[numGroups];
-            
+            return groupsOfIndices;
+        }
+
+        /// <summary>
+        /// This takes in a list of ints, and partitions them into numGroups partitions,
+        /// e.g., partition 1 = [0, 4, 8...], partition 2 = [1, 5, 9...], etc.
+        /// </summary>
+        /// <returns>A list containing numGroups partitions (lists of ints) </returns>
+        static List<List<int>> DivideListIntoGroups(List<int> list, int numGroups)
```

Comment:
there was a previously less complicated method here. did it not function correctly?

Reply:
previously, the method took sequential chunks from each list. e.g., [1,2,3,4,5,6,7,8] would return [[1, 2], [3, 4], [5, 6], [7, 8]].

This function would return [[1, 5], [2, 6], [3, 7], [4, 8]].